### PR TITLE
[KED-3065] Why Kedro component

### PR DESCRIPTION
**Pull request by tynandebold**
- This adds the Why Kedro? component.
- I've refactored Rashida's work a little bit and pulled in some of the necessary setup from the Hero PR.
- I've added Framer Motion and used one of [their examples](https://www.framer.com/docs/examples/#shared-layout-animations) to build the tabs. This'll give us a great baseline for future animation. I think the designers will be happy.
- Do note that the development of the tabs doesn't match up 1-to-1 with the design. I've flagged this to Mackay. What they designed had much shorter text strings. The actual copy it much longer. He'll play with it and get back to us.
- Also note with the tabs on smaller screens that the scrolling isn't identical either. We need to have a short sync with the designers to see what exactly they want this behavior to be.
- Don't pay too much attention to the space above and below the component either. That will be ironed out when we start composing the page.

### CODE REVIEW

**Review by rashidakanchwala — APPROVED** [review]
Looks great and works well -- a lot more simpler than my initial approach .. thanks Tynan !

I just had one question -- i assumed the media component was specific for each tab. But in the code it just sits under the tabs!

**Comment by tynandebold** [comment]
> Looks great and works well -- a lot more simpler than my initial approach .. thanks Tynan !
> 
> I just had one question -- i assumed the media component was specific for each tab. But in the code it just sits under the tabs!

I thought the same and asked Mackay about it. He said that actually the media element will be specific to this _whole section_, not each tab, as you said above. I also thought the same as you did, so it was good I asked him.

File: components/why-kedro/why-kedro.module.scss [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+
+  @include breakpoint(small) {
+    margin: 0 auto 1.375rem;
+    padding-left: 1rem;
+    width: 100%;
+  }
+}
+
+.list,
+.listItem {
```
Comment (studioswong) [inline components/why-kedro/why-kedro.module.scss:31]: as discussed in our last PR, given that we are still using sass, moreover with how we declared our variables, let's follow the hyphen-delimited approach to naming our scss casses.

Comment (tynandebold) [reply components/why-kedro/why-kedro.module.scss:31]: It does seem we can keep it how it's written here based on [Next's own documentation](https://nextjs.org/docs/basic-features/built-in-css-support#sass-variables). I'm trying to follow their example.

Were you able to look at the rest of the PR? Would love if you can please approve/request changes before the EOD day if possible!

**Review by studioswong — CHANGES_REQUESTED** [review]
Overall the tabs works well, the tab animation looks good - the framer library is a good choice.

Some quick observations I notice that would be good to address in this PR: 
The current tab panel on mobile breakpoint cuts into the edge a compared to the design:

image

image

I know we don't have the design for iPad breakpoints, but testing the responsiveness of the page, the tabbed panel also cuts right towards the edge of the screen, which could be resolved by applying an overall margin for the content under `index.tsx`. 

image

**Comment by tynandebold** [comment]
Good catches. I've fixed the image not having any spacing and believe I fixed the other point you mentioned about the tabs cutting to the edge of the screen (though they are supposed to do that on narrow screen, on the right side). Hope I got it right.

File: components/why-kedro/why-kedro.module.scss (hunk 1 above)
Comment (studioswong) [reply components/why-kedro/why-kedro.module.scss:31]: I understand we can keep it, but that doesn't mean we should follow it - I still think there is merit in maintaining the consistency of the css naming convention we have employed in the team, moreover it being in line with the variable naming that was set up in `variables.scss`. 

I had put in my review earlier, let me know if there's anything unclear. I can come back to review this later today once the requested changes are made to help move this forward.

Comment (tynandebold) [reply components/why-kedro/why-kedro.module.scss:31]: You can review it again now. I pushed some changes ~5 minutes ago for the style fixes you caught.

Comment (tynandebold) [reply components/why-kedro/why-kedro.module.scss:31]: I'm trying to draw a distinction between the code and product of Kedro-Viz and this Kedro website code and project. I don't believe all conventions in one need to be transferred over to the other.

If the logic is that we should align with the variable naming in `variables.scss` then I think that means we actually **do** want to use camelCase. This is the pattern and best practice outlined by the framework.

Next has this:

```scss
$primary-color: #64FF00

:export {
  primaryColor: $primary-color
}
```

and then they reference it like so:

```jsx
<Layout color={variables.primaryColor} />
```

Is that what we're doing as well?

**Comment by studioswong** [comment]
Thanks for adding in the margin for the tab, looks better now for the mobile view. I can also see the hero components that was merged in from the last PR, which is great. 

I notice that there is a huge gap between the Hero section and the Why Kedro section which is different from the design as seen below:

image

image

I also notice that the margin for the larger responsive pages is still bleeding into the edge - it would be great if we can add a margin around the tabbed panel. 

image

**Comment by tynandebold** [comment]
Thanks so much.

Added:
- Margin around the tabs on the middle-width screen size.
- Smaller spacing between the sections on small screen sizes.

File: components/why-kedro/why-kedro.module.scss (hunk 1 above)
Comment (studioswong) [reply components/why-kedro/why-kedro.module.scss:35]: hhmm I see - personally I love keeping style classes as kebab case as I often see in most FE projects, but I understand if this is the norm for next.js. In this case it's fine and it's no fuss - we can keep it as is then.

**Review by studioswong — APPROVED** [review]
Thanks for addressing the edge issues 😄

**Pull request merged**